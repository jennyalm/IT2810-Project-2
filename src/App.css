
.App{
    display: grid;
    grid-template-columns: [left] 8fr [mid] 2fr [right];
    grid-template-rows: [top] 1fr [mid] 6fr [bottom];
}

.headerDiv{
    grid-column-start: left;
    grid-column-end: right;
    grid-row-start: top;
    grid-row-end: mid;
    height: 100%;

    background-color: whitesmoke;

    letter-spacing: 1px;
    font-size: 170%;
    color: #323232;
    text-align: center;
    font-family: "Lato", sans-serif;
    font-weight: lighter;
}


.App-header {
    grid-column-start: left;
    grid-column-end: right;
    grid-row-start: top;
    grid-row-end: mid;
    height: 100%;

    background-color: whitesmoke;

    letter-spacing: 1px;
    font-size: 170%;
    color: #323232;
    text-align: center;
    font-family: "Lato", sans-serif;
    font-weight: lighter;

}

.knapp{
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    border: solid #ccc;
    border-width: 0 0 1px 0px;


}

.knapp1{
    background: white;
    color: #424242;
    border-width: 0px 0px 0px 0px;
    margin-bottom: 10px;
    margin-top: 10px;
    height: 5vh;
    width: auto;
    font-size: 1vw;
}

.knapp1:hover{
    background: #b7b7b7;
    color: #323232;
    font-family: "Lato", sans-serif;
}

.knapp2{
    margin-top: 10px;
    margin-bottom: 20px;
    background: white;
    color: #424242;
    border-width: 0px 0px 0px 0px;
    font-family: "Lato", sans-serif;
    height: 5vh;
    width: auto;
    font-size: 1vw;
}

.knapp2:hover{
    background: #b7b7b7;
    color: #323232;
    font-family: "Lato", sans-serif;
}

.tab-align{
    grid-column-start: left;
    grid-column-end: mid;
    grid-row-start: mid;
    grid-row-end: bottom;
    fontFamily: sans-serif;
    textAlign: center;
}

.settings{
    grid-column-start: mid;
    grid-column-end: right;
    grid-row-start: mid;
    grid-row-end: bottom;
    padding-top: 12%;
    padding-left: 20%;
    padding-right: 20%;
    letter-spacing: 1px;
}



.App-logo {
    height: 50vmin;
    border: solid #282c34 3px;

}

.categories{
    border: solid #ccc;
    border-width: 0 0px 0 1px;
    padding-left: 20px;
    padding-right: 20px;

}


.tabs {
    margin-left: 10%;
}

@media only screen and (max-width: 600px){
    body{
        font-size: 2vw;
    }
    .App{
        display: grid;
        grid-template-columns: [left] 5fr [right];
        grid-template-rows: [top] .2fr [top-mid] .5fr [bottom-mid] 3fr [bottom];
        max-width: 375px;
        padding-left: 20px;
        padding-right: 20px;
    }

    .App-header{
        text-align: left;
        margin-bottom: 20px;
    }

    .headerDiv{
        grid-row-start: top;
        grid-row-end: top-mid;
        grid-column-start: left;
        grid-column-end: right;
        text-align: left;
        margin-left: 5px;
        font-size: 1.5vh;
    }

    .appContent{
        grid-row-start: bottom-mid;
        grid-row-end: bottom;
        grid-column-start: left;
        grid-column-end: right;
        width: auto;
        height: 100vh;
        font-size: 1vh;

    }

    .settings{
        grid-row-start: top-mid;
        grid-row-end: bottom-mid;
        padding: 0;
        grid-column-start: left;
        grid-column-end: right;
    }

    .categories{
        display: grid;
        grid-template-columns: [left-end] 1fr [left-mid] 1fr [mid-end] 1fr [right-mid];
        grid-template-rows: [top-end] 1fr [bottom-end] 1fr [bottom-end2];
        padding: 0;
        border-width: 0 0 1px 0;

    }

    .bildeSetting{
        grid-column-start: left-end;
        grid-column-end: left-mid;
        grid-row-start: top-end;
        grid-row-end: bottom-end;
    }

    .lydSetting{
        grid-column-start: left-mid;
        grid-column-end: mid-end;
        grid-row-start: top-end;
        grid-row-end: bottom-end;
    }

    .tekstSetting{
        grid-column-start: mid-end;
        grid-column-end: right-mid;
        grid-row-start: top-end;
        grid-row-end: bottom-end;
    }

    .storageSetting{
        grid-column-start: left-end;
        grid-column-end: right-end;
        grid-row-start: bottom-end;
        grid-row-end: bottom-end2;
    }









    .tabs{
        margin-left: 0;
    }

    .knapp{
        grid-column-start: left-end;
        grid-column-end: right-end;
        grid-row-start: bottom-end;
        grid-row-end: bottom-end2;

        padding-top: 1vh;
        display: grid;
        grid-template-columns: [venstre] 1fr [midt] 1fr [høyre];
        grid-template-rows: [toppen] 1fr [bunn];
        border-width: 0 0 0 0;

    }
    .knapp1{
        grid-column-start:venstre;
        grid-column-end: midt;
        grid-row-start: toppen;
        grid-row-end: bunn;


        height: 3vh;
        width: auto;
        margin: 10px;
        padding: 0;
        font-size: 2vw;
    }

    .knapp2{
        grid-column-start:midt;
        grid-column-end: høyre;
        grid-row-start: toppen;
        grid-row-end: bunn;


        height: 3vh;
        width: auto;
        margin: 10px;
        padding: 0;
        font-size: 2vw;
    }
}

@media only screen and (min-width: 600px) and (max-width: 850px){
    .appContent{
        width: auto;
    }

}
