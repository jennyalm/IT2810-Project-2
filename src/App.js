import React from 'react';
import './App.css';


import Header from "./components/Header";
import TabContent from "./components/TabContent";
import MediaSettings from "./components/MediaSettings";


class App extends React.Component {
    constructor(){
        super()
        this.state = {
            tabCategory: "kombinasjon1",
            pictureCategory: "dessert",
            imageName: "cake.svg",
            textCategory: "1",
            soundCategory: "electronic",
            soundName: "electronic1.mp3",
            favorite: [],
            clicks: sessionStorage.clickcount
        }

        // binds the functions to this class (App)
        this.handleNewPicture = this.handleNewPicture.bind(this)
        this.handleClick = this.handleClick.bind(this)
        this.handleChangeOnPage = this.handleChangeOnPage.bind(this)
        this.handleNewText = this.handleNewText.bind(this)
        this.handleNewSound = this.handleNewSound.bind(this)
        this.displayFavorite = this.displayFavorite.bind(this)
        this.saveFavorite = this.saveFavorite.bind(this)
        this.handleSoundChange = this.handleSoundChange.bind(this)
        this.clickCounter = this.clickCounter.bind(this)

    }

    // handles det changes to the picture given the different states from handleClick og handleNewPicture
    // value is this.state.pictureCategory and tab is this.state.tabCategory. Have to send them in as params because setState in handleNewPicture and handleClick are to slow to keep up.
    handleChangeOnPage(value, tab){
        let dessert = ["cake.svg", "icecream.svg", "pie.svg", "pudding.svg"]
        let drinks = ["beer.svg", "juice.svg", "soda.svg", "wine.svg"]
        let food = ["burger.svg", "hotdog.svg", "pizza.svg", "taco.svg"]
        let imgName = ""
        if(value === "dessert"){
            if(tab === "kombinasjon1"){
                imgName = dessert[0]
            } else if(tab === "kombinasjon2"){
                imgName = dessert[1]
            } else if(tab === "kombinasjon3"){
                imgName = dessert[2]
            } else{
                imgName = dessert[3]
            }
        } else if(value === "drinks"){
            if(tab === "kombinasjon1"){
                imgName = drinks[0]
            } else if(tab === "kombinasjon2"){
                imgName = drinks[1]
            } else if(tab === "kombinasjon3"){
                imgName = drinks[2]
            } else{
                imgName = drinks[3]
            }
        } else if(value === "food"){
            if(tab === "kombinasjon1"){
                imgName = food[0]
            } else if(tab === "kombinasjon2"){
                imgName = food[1]
            } else if(tab === "kombinasjon3"){
                imgName = food[2]
            } else{
                imgName = food[3]
            }
        }
        this.setState((prevState) => {
            return{
                imageName: imgName
            }
        })
    }

    // handles det changes to the sound given the different states from handleClick og handleNewSound
    // value is this.state.soundCategory and tab is this.state.tabCategory. Have to send them in as params because setState in handleNewSound and handleClick are to slow to keep up.
    handleSoundChange(value, tab){
        let electronic = ["electronic1.mp3", "electronic2.mp3", "electronic3.mp3", "electronic4.mp3"]
        let guitar = ["guitar1.mp3", "guitar2.mp3", "guitar3.mp3", "guitar4.mp3"]
        let piano = ["piano1.mp3", "piano2.mp3", "piano3.mp3", "piano4.mp3"]
        let soName = ""
        if(value === "electronic"){
            if(tab === "kombinasjon1"){
                soName = electronic[0]
            } else if(tab === "kombinasjon2"){
                soName = electronic[1]
            } else if(tab === "kombinasjon3"){
                soName = electronic[2]
            } else{
                soName = electronic[3]
            }
        } else if(value === "guitar"){
            if(tab === "kombinasjon1"){
                soName = guitar[0]
            } else if(tab === "kombinasjon2"){
                soName = guitar[1]
            } else if(tab === "kombinasjon3"){
                soName = guitar[2]
            } else{
                soName = guitar[3]
            }
        } else if(value === "piano"){
            if(tab === "kombinasjon1"){
                soName = piano[0]
            } else if(tab === "kombinasjon2"){
                soName = piano[1]
            } else if(tab === "kombinasjon3"){
                soName = piano[2]
            } else{
                soName = piano[3]
            }
        }

        this.setState((prevState) => {
            return{
                soundName: soName
            }
        })
    }

    // changes tabCategory depending on which tab you click
    handleClick(label){
        this.setState(prevState => {
            return{
                tabCategory: label
            }
        })
        // since pictureCategory and soundCategory updates here, we can use the state.
        this.handleChangeOnPage(this.state.pictureCategory, label)
        this.handleSoundChange(this.state.soundCategory, label)
    }


    // runs whenever a new category of pictures is clicked, changes the state to the new one, and then runs handleChangeOnPage which displays the new image
    handleNewPicture(event){
        const {value, name} = event.target

        this.setState((prevState) => {
            return{
                [name]: value,
            }
        });
        this.handleChangeOnPage(value, this.state.tabCategory)
    }


    // runs whenever a new category of text is clicked, changes the state to the new one, and updates the text on the page. (the text updates because of the componentDidUpdate in FetchText)
    handleNewText(event){
        const {value, name} = event.target

        this.setState((prevState) => {
            return{
                [name]: value,
            }
        });
    }

    // runs whenever a new category of sound is clicked, changes the state to the new one, and then runs handleSoundChange to display the new sound
    handleNewSound(event){

        const {value, name} = event.target
        this.setState({[name]: value});
        this.handleSoundChange(value, this.state.tabCategory)

    }

    // gets the states which is saved in localStorage and displays them to the site. Also runs the clickCounter.
    displayFavorite(){
        this.clickCounter();
        this.setState(prevState => {
            return{
                tabCategory: localStorage.getItem("tabCategory"),
                pictureCategory: localStorage.getItem("pictureCategory"),
                imageName: localStorage.getItem("imageName"),
                textCategory: localStorage.getItem("textCategory"),
                soundCategory: localStorage.getItem("soundCategory"),
                soundName: localStorage.getItem("soundName")
            }
        })
        this.handleClick(localStorage.getItem("tabCategory"))
        this.handleChangeOnPage(localStorage.getItem("pictureCategory"), localStorage.getItem("tabCategory"))
        this.handleSoundChange(localStorage.getItem("soundCategory"), localStorage.getItem("tabCategory"))

    }

    // saves the current states to localStorage and runs clickCounter.
    saveFavorite(){
        this.clickCounter();
        this.setState(prevState => {
            return{
                favorite: [
                    localStorage.setItem("tabCategory", prevState.tabCategory),
                    localStorage.setItem("pictureCategory", prevState.pictureCategory),
                    localStorage.setItem("imageName", prevState.imageName),
                    localStorage.setItem("textCategory", prevState.textCategory),
                    localStorage.setItem("soundCategory", prevState.soundCategory),
                    localStorage.setItem("soundName", prevState.soundName)
                ]
            }
        })
    }

    // updates sessionStorage.clickcount with +1 and updates the this.state.clicks
    clickCounter(){
        if (sessionStorage.clickcount) {
           sessionStorage.clickcount = Number(sessionStorage.clickcount) + 1;
        } else {
            sessionStorage.clickcount = 1;
        }
        this.setState({clicks: sessionStorage.clickcount})
    }


    // renders everything, divided up in 3 parts.
    // 1. Header which displays the header
    // 2. TabContent which displays everything inside the tabs, and the tabs themselves.
    // 3. MediaSettings which displays the different settings, such as the category radio buttons and the save/show favorite buttons
    render(){
        return (
            <div className="App">
                <div className="headerDiv">
                    <Header />
                </div>
                <div className="appContent">
                    <TabContent imageName={this.state.imageName}
                                handleClick={this.handleClick}
                                tabCategory={this.state.tabCategory}
                                textCategory={this.state.textCategory}
                                pictureCategory={this.state.pictureCategory}
                                soundCategory={this.state.soundCategory}
                                soundName={this.state.soundName}

                    />
                </div>
                <div className={"settings"}>
                    <MediaSettings pictureCategory={this.state.pictureCategory}
                                   handleNewPicture={this.handleNewPicture}
                                   soundCategory={this.state.soundCategory}
                                   handleNewSound={this.handleNewSound}
                                   textCategory={this.state.textCategory}
                                   handleNewText={this.handleNewText}
                                   saveFavorite={this.saveFavorite}
                                   displayFavorite={this.displayFavorite}
                                   clicks={ this.state.clicks}

                    />
                </div>
                <br />
            </div>
                )
    }

}

export default App;
