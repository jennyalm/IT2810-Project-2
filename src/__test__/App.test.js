import React from 'react'
import renderer from 'react-test-renderer'

import App from '../App';

test('App snapshot test', () => {



    const tree = renderer.create( <App/> ).toJSON();

    expect(tree).toMatchSnapshot();
});