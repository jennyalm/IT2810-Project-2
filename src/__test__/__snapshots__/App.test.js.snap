// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      Webutvikling prosjekt 2 - Håkon, Jenny og Jakob
    </p>
  </header>
  <div
    className="tabs"
  >
    <ol
      className="tab-list"
    >
      <li
        className="tab-list-item tab-list-active"
        onClick={[Function]}
      >
        kombinasjon1
      </li>
      <li
        className="tab-list-item"
        onClick={[Function]}
      >
        kombinasjon2
      </li>
      <li
        className="tab-list-item"
        onClick={[Function]}
      >
        kombinasjon3
      </li>
      <li
        className="tab-list-item"
        onClick={[Function]}
      >
        kombinasjon4
      </li>
    </ol>
    <div
      className="tab-content"
    >
      <div
        className="Tab-Content"
      >
        <div
          className="pic"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
          />
        </div>
        <div
          className="info"
        >
          <div
            className="info-text"
          >
            <p />
          </div>
          <div
            className="play-button"
          >
            <div>
              <audio
                controls={true}
              >
                <source
                  src="./Sound/electronic/electronic1.mp3"
                  type="audio/mpeg"
                />
              </audio>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="settings"
  >
    <div
      className="categories"
    >
      <p
        className="bilde"
      >
        Bilde:
      </p>
      <form>
        <label>
          <input
            checked={true}
            name="pictureCategory"
            onChange={[Function]}
            type="radio"
            value="dessert"
          />
          Dessert
        </label>
      </form>
      <form>
        <label>
          <input
            checked={false}
            name="pictureCategory"
            onChange={[Function]}
            type="radio"
            value="drinks"
          />
          Drikke
        </label>
      </form>
      <form>
        <label>
          <input
            checked={false}
            name="pictureCategory"
            onChange={[Function]}
            type="radio"
            value="food"
          />
          Mat
        </label>
      </form>
      <p>
        Lyd:
      </p>
      <form>
        <label>
          <input
            checked={true}
            name="soundCategory"
            onChange={[Function]}
            type="radio"
            value="electronic"
          />
          Elektronisk
        </label>
      </form>
      <form>
        <label>
          <input
            checked={false}
            name="soundCategory"
            onChange={[Function]}
            type="radio"
            value="guitar"
          />
          Gitar
        </label>
      </form>
      <form>
        <label>
          <input
            checked={false}
            name="soundCategory"
            onChange={[Function]}
            type="radio"
            value="piano"
          />
          Piano
        </label>
      </form>
      <p>
        Tekst:
      </p>
      <form>
        <label>
          <input
            checked={true}
            name="textCategory"
            onChange={[Function]}
            type="radio"
            value="1"
          />
          Dikt
        </label>
      </form>
      <form>
        <label>
          <input
            checked={false}
            name="textCategory"
            onChange={[Function]}
            type="radio"
            value="2"
          />
          Haiku
        </label>
      </form>
      <form>
        <label>
          <input
            checked={false}
            name="textCategory"
            onChange={[Function]}
            type="radio"
            value="3"
          />
          Vits
        </label>
      </form>
      <br />
      <div
        className="knapp"
      >
        <button
          className="knapp1"
          onClick={[Function]}
        >
          Vis favoritt
        </button>
        <button
          className="knapp2"
          onClick={[Function]}
        >
          Lagre favoritt
        </button>
        <p>
          Du har trykket: 
           ganger
        </p>
      </div>
    </div>
  </div>
  <div>
    <h4>
      Litt data for å holde oversikt og sjekke at ting funker: 
    </h4>
    <h5>
      Tab Category: 
      kombinasjon1
    </h5>
    <h5>
      Picture Category: 
      dessert
    </h5>
    <h5>
      Sound Category: 
      electronic
    </h5>
    <h5>
      Text Category: 
      1
    </h5>
  </div>
  <br />
</div>
`;
